# SQL Business Analysis

## Objective

The objective of this phase is to use SQL queries to answer key business questions related to Rossmann's sales performance, customer behavior, promotional effectiveness, and store operations.

The insights generated will support business decision-making and prepare the dataset for predictive modeling.


In [10]:
import pandas as pd
import sqlite3

df = pd.read_csv(
    "/content/cleaned_sales.csv",
    dtype={"StateHoliday": str}
)
df.head()
conn = sqlite3.connect("rossmann.db")

df.to_sql(
    "cleaned_sales",
    conn,
    if_exists="replace",
    index=False
)
query = """
SELECT *
FROM cleaned_sales
LIMIT 5;
"""

pd.read_sql(query, conn)

,Store,DayOfWeek,Date,Sales,Customers,Open,Promo,StateHoliday,SchoolHoliday,StoreType,...,Promo2SinceWeek,Promo2SinceYear,PromoInterval,Year,Month,Week,Day,Quarter,DayName,Weekend
0,1,5,2015-07-31,5263,555,1,1,0,1,c,...,0.0,0.0,None,2015,7,31,31,3,Friday,0
1,2,5,2015-07-31,6064,625,1,1,0,1,a,...,13.0,2010.0,"Jan,Apr,Jul,Oct",2015,7,31,31,3,Friday,0
2,3,5,2015-07-31,8314,821,1,1,0,1,a,...,14.0,2011.0,"Jan,Apr,Jul,Oct",2015,7,31,31,3,Friday,0
3,4,5,2015-07-31,13995,1498,1,1,0,1,c,...,0.0,0.0,None,2015,7,31,31,3,Friday,0
4,5,5,2015-07-31,4822,559,1,1,0,1,a,...,0.0,0.0,None,2015,7,31,31,3,Friday,0


# 1. Revenue KPI Analysis

## Business Question

What is the total revenue generated across all Rossmann stores?

In [11]:
query = """
SELECT
    SUM(Sales) AS Total_Revenue
FROM cleaned_sales;
"""

pd.read_sql(query, conn)

,Total_Revenue
0,5873180623


### Observation

Rossmann generated a total revenue of **₹5.87 billion** during the observed period, demonstrating the large scale of its retail operations across all stores.

### Business Insight

The overall revenue establishes a baseline KPI for measuring business performance. Future analyses can compare this value across store types, promotional periods, holidays, and seasons to identify revenue drivers.

### Recommendation

Management should monitor this KPI monthly and annually to evaluate business growth and assess the effectiveness of strategic initiatives such as promotions and store expansions.

### Business Impact

Tracking total revenue enables executives to evaluate overall financial performance, support strategic planning, and measure long-term business growth.

#2.Average Daily Revenue

##Business Question

What is the average revenue generated per day?



In [12]:
query = """
SELECT
    ROUND(AVG(Sales),2) AS Average_Daily_Revenue
FROM cleaned_sales;
"""

pd.read_sql(query, conn)

,Average_Daily_Revenue
0,5773.82


### Observation

Each sales record generated an average revenue of **₹5,773.82**, providing a useful benchmark for evaluating store-level performance.

### Business Insight

This metric represents the average sales generated per store-day observation. Stores consistently below this benchmark may require operational improvements or targeted promotional strategies.

### Recommendation

Use ₹5,773.82 as a benchmark to identify underperforming stores and evaluate the impact of marketing campaigns and pricing strategies.

### Business Impact

Monitoring average sales helps improve operational efficiency and supports data-driven performance management across stores.

#3.Total Number of Transactions

##Business Question

How many sales records exist?



In [13]:
query = """
SELECT
    COUNT(*) AS Total_Transactions
FROM cleaned_sales;
"""

pd.read_sql(query, conn)

,Total_Transactions
0,1017209


### Observation

The dataset contains **1,017,209 sales records**, representing over one million store-day observations collected during the study period.

### Business Insight

The large volume of data provides a strong foundation for reliable business analysis and predictive sales forecasting.

### Recommendation

Leverage the richness of the dataset to identify sales trends, customer behavior, and seasonal demand patterns.

### Business Impact

A large historical dataset improves confidence in business insights and enhances the accuracy of forecasting models.

#4.Revenue by Store Type

##Business Question

Which store type generates the highest revenue?


In [16]:
query = """
SELECT
    StoreType,
    printf('%,.0f', SUM(Sales)) AS Total_Revenue
FROM cleaned_sales
GROUP BY StoreType
ORDER BY SUM(Sales) DESC;
"""

pd.read_sql(query, conn)

,StoreType,Total_Revenue
0,a,3165334859
1,d,1765392943
2,c,783221426
3,b,159231395


### Observation

Store Type **A** generated the highest revenue (₹3.17 billion), accounting for more than half of the company's total sales. Store Type **D** ranked second with ₹1.77 billion, while Store Type **B** contributed the least revenue.

### Business Insight

Store type has a significant impact on revenue generation. The substantial difference between store types suggests that factors such as location, assortment, customer demographics, or store size influence sales performance.

### Recommendation

Investigate the operational practices of high-performing Store Type A locations and evaluate whether successful strategies can be adapted for lower-performing store types.

### Business Impact

Understanding revenue by store type enables management to optimize investment decisions, improve store formats, and maximize long-term profitability.

#5.Top 10 Revenue Generating Stores

##Business Question

Which stores contribute the highest revenue?

In [15]:
query = """
SELECT
    Store,
    ROUND(SUM(Sales),2) AS Total_Revenue
FROM cleaned_sales
GROUP BY Store
ORDER BY Total_Revenue DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,Store,Total_Revenue
0,262,19516842.0
1,817,17057867.0
2,562,16927322.0
3,1114,16202585.0
4,251,14896870.0
5,513,14252406.0
6,788,14082141.0
7,733,14067158.0
8,383,13489879.0
9,756,12911782.0


### Observation

Store **262** generated the highest revenue (₹19.52 million), followed by Stores **817** and **562**. The top-performing stores consistently generated significantly higher revenue than the network average.

### Business Insight

Revenue is concentrated among a relatively small number of high-performing stores. These locations likely benefit from favorable factors such as strong customer demand, effective promotions, strategic locations, or superior operational management.

### Recommendation

Conduct a detailed performance review of the top 10 stores to identify best practices related to inventory management, staffing, promotions, and product assortment. Apply successful strategies to similar stores where appropriate.

### Business Impact

Replicating operational practices from the highest-performing stores can improve sales across the broader store network and contribute to sustainable revenue growth.